packages/webview_flutter/webview_flutter_wkwebview/lib/src/foundation/foundation.dart

```diff
@@ -235,6 +235,42 @@ class NSHttpCookie {
   final Map<NSHttpCookiePropertyKey, Object> properties;
 }
 
+/// An object that represents the location of a resource, such as an item on a
+/// remote server or the path to a local file.
+///
+/// See https://developer.apple.com/documentation/foundation/nsurl?language=objc.
+class NSUrl extends NSObject {
+  /// Instantiates a [NSUrl] without creating and attaching to an instance
+  /// of the associated native class.
+  ///
+  /// This should only be used outside of tests by subclasses created by this
+  /// library or to create a copy for an [InstanceManager].
+  @protected
+  NSUrl.detached({super.binaryMessenger, super.instanceManager})
+      : _nsUrlHostApi = NSUrlHostApiImpl(
+          binaryMessenger: binaryMessenger,
+          instanceManager: instanceManager,
+        ),
+        super.detached();
+
+  final NSUrlHostApiImpl _nsUrlHostApi;
+
+  /// The URL string for the receiver as an absolute URL. (read-only)
+  ///
+  /// Represents [NSURL.absoluteString](https://developer.apple.com/documentation/foundation/nsurl/1409868-absolutestring?language=objc).
+  Future<String?> getAbsoluteString() {
+    return _nsUrlHostApi.getAbsoluteStringFromInstances(this);
+  }
+
+  @override
+  NSObject copy() {
+    return NSUrl.detached(
+      binaryMessenger: _nsUrlHostApi.binaryMessenger,
+      instanceManager: _nsUrlHostApi.instanceManager,
+    );
+  }
+}
```

Comment:
Not something that needs action, but to file for reference as we evaluate this approach: I think `NSURL` is a good example of a place where hewing as close as possible to the native APIs is actually more trouble than it's worth (until/unless we have auto-generated language projections). It's essentially just a string with utility methods, and we have very similar functionality in Dart already in the form of `Uri`. In practice, if we passed strings over the channel instead of objects, and converted them to `NSURL` and `Uri` on the respective language sides, it would be simpler since it would eliminate object tracking for URLs, and work just as well in almost all cases. (The only example I can think of where it wouldn't would be things that can be expressed as `NSURL`s but not `Uri`s, but that's *very* rare.)